test/extensions/filters/common/rbac/matchers_test.cc

```diff
@@ -699,6 +699,263 @@ TEST(MetadataMatcher, SourcedMetadataMatcher) {
+  info.downstream_connection_info_provider_->setLocalAddress(addr1);
+  checkMatcher(PortRangeMatcher(range), true, conn, headers, info);
+
+  // Test port at end of range - 1
+  Envoy::Network::Address::InstanceConstSharedPtr addr2 =
+      Envoy::Network::Utility::parseInternetAddressNoThrow("1.2.3.4", 65534, false);
+  info.downstream_connection_info_provider_->setLocalAddress(addr2);
+  checkMatcher(PortRangeMatcher(range), true, conn, headers, info);
+
+  // Test port exactly at end (should be false as range is exclusive at end)
+  Envoy::Network::Address::InstanceConstSharedPtr addr3 =
+      Envoy::Network::Utility::parseInternetAddressNoThrow("1.2.3.4", 65535, false);
+  info.downstream_connection_info_provider_->setLocalAddress(addr3);
+  checkMatcher(PortRangeMatcher(range), false, conn, headers, info);
+}
+
+TEST(PathMatcher, HeaderEdgeCases) {
+  NiceMock<Server::Configuration::MockServerFactoryContext> context;
+  Envoy::Http::TestRequestHeaderMapImpl headers;
+  envoy::type::matcher::v3::PathMatcher matcher;
+
+  // Test that path matcher matches exact paths
+  matcher.mutable_path()->mutable_safe_regex()->mutable_google_re2();
+  matcher.mutable_path()->mutable_safe_regex()->set_regex("^/$");
+  headers.setPath("/");
+  checkMatcher(PathMatcher(matcher, context), true, Envoy::Network::MockConnection(), headers);
+
+  // Test path with query parameters - regex should match path part only
+  matcher.mutable_path()->mutable_safe_regex()->set_regex("^/path$");
+  headers.setPath("/path?param=value");
+  checkMatcher(PathMatcher(matcher, context), true, Envoy::Network::MockConnection(), headers);
+  headers.setPath("/different?param=value");
+  checkMatcher(PathMatcher(matcher, context), false, Envoy::Network::MockConnection(), headers);
+
+  // Test URL encoded paths - should match the encoded form
+  matcher.mutable_path()->mutable_safe_regex()->set_regex("^/path%20with%20spaces$");
+  headers.setPath("/path%20with%20spaces");
+  checkMatcher(PathMatcher(matcher, context), true, Envoy::Network::MockConnection(), headers);
+  headers.setPath("/path with spaces"); // Not URL encoded
+  checkMatcher(PathMatcher(matcher, context), false, Envoy::Network::MockConnection(), headers);
+}
+
+TEST(HeaderMatcher, MultipleHeaderValues) {
+  NiceMock<Server::Configuration::MockServerFactoryContext> factory_context;
+  envoy::config::route::v3::HeaderMatcher config;
+  config.set_name("multi-header");
+  config.mutable_string_match()->set_exact("value1");
+
+  Envoy::Http::TestRequestHeaderMapImpl headers;
+  Envoy::Http::LowerCaseString header_name("multi-header");
+  headers.setReference(header_name, "value1");
+
+  RBAC::HeaderMatcher matcher(config, factory_context);
+  checkMatcher(matcher, true, Envoy::Network::MockConnection(), headers);
+
+  // Test with non-matching value
+  headers.remove(header_name);
+  headers.setReference(header_name, "value2");
+  checkMatcher(matcher, false, Envoy::Network::MockConnection(), headers);
+}
```

Comment:
Where is the multiple header values matching?

Reply:
@wbpcode Thanks for the feedback! I tried adding some cases.